src/common/greptimedb-telemetry/src/lib.rs

```diff
@@ -0,0 +1,371 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::env;
+use std::io::ErrorKind;
+use std::path::PathBuf;
+use std::time::Duration;
+
+use common_runtime::error::{Error, Result};
+use common_runtime::{BoxedTaskFunction, RepeatedTask, Runtime, TaskFunction};
+use common_telemetry::debug;
+use once_cell::sync::Lazy;
+use reqwest::{Client, Response};
+use serde::{Deserialize, Serialize};
+
+pub const TELEMETRY_URL: &str = "https://api-preview.greptime.cloud/db/otel/statistics";
+
+// Getting the right path when running on windows
+static TELEMETRY_UUID_FILE_NAME: Lazy<PathBuf> = Lazy::new(|| {
+    let mut path = PathBuf::new();
+    path.push(env::temp_dir());
```

Comment:
problem: I wonder if it works. According to the docs, we need to create a unique file name.

https://doc.rust-lang.org/stable/std/env/fn.temp_dir.html

> The temporary directory may be shared among users, or between processes with different privileges; thus, the creation of any files or directories in the temporary directory must use a secure method to create a uniquely named file. Creating a file or directory with a fixed or predictable name may result in “insecure temporary file” security vulnerabilities. Consider using a crate that securely creates temporary files or directories.

Consider using [tempfile](https://docs.rs/tempfile/latest/tempfile/)

Reply:
We need to write a file with a known location so that we can find it later and get the uuid, but it's true that it could be rewritten by another process. It doesn't matter, the consequence is that the uuid will be regenerated next time.

Reply:
@paomian @evenyag I will do this job while refactoring `data_home` option.